core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2522,6 +2522,20 @@ public AlluxioURI getPath(long fileId) throws FileDoesNotExistException {
     return mInodeTree.getPinIdSet();
   }
 
+  @Override
+  public List<String> getPinnedFilePaths() {
+    List<String> pinnedFiles = new ArrayList<>();
+    for (long pinId : getPinIdList()) {
+      try {
+        String pinFilePath = getPath(pinId).getPath();
```

Comment:
here is one concern: when a user "foo" invokes `getPinnedFilePaths`, does this user has the permission to list all the pinned files? In worst case, there might be files in paths user "foo" does not have permission to list.